pkg/test/runner/runner.go

```diff
@@ -262,7 +255,7 @@ func (r *Runner) compareResult(exitErr error, tmpPkgPath, resultsPath string) er
 		return fmt.Errorf("actual exit code %d doesn't match expected %d", exitCode, r.testCase.Config.ExitCode)
 	}
 
-	if exitCode != 0 {
+	if exitCode != 0 && expected.Results != "" {
```

Comment:
This doesn't handle the case that the fn exit with 0 but it output some structured results (e.g. warning and info).

Reply:
Updated and now we will always compare results if there is any.